packages/@react-aria/calendar/src/useCalendarCell.ts

```diff
@@ -283,17 +283,27 @@ export function useCalendarCell(props: AriaCalendarCellProps, state: CalendarSta
 
   // Focus the button in the DOM when the state updates.
   useEffect(() => {
+    let raf;
     if (isFocused && ref.current) {
       focusWithoutScrolling(ref.current);
 
       // Scroll into view if navigating with a keyboard, otherwise
       // try not to shift the view under the user's mouse/finger.
-      // Only scroll the direct scroll parent, not the whole page, so
-      // we don't scroll to the bottom when opening date picker popover.
+      // If in a overlay, scrollIntoViewport will only cause scrolling
+      // up to the overlay scroll body to prevent overlay shifting
       if (getInteractionModality() !== 'pointer') {
-        scrollIntoView(getScrollParent(ref.current) as HTMLElement, ref.current);
+        // If this calendar is in a popover, we need to wait till it is fully positioned before attempting to
+        // scroll it into view, otherwise it won't be brought into view properly
+        raf = requestAnimationFrame(() => {
```

Comment:
I think we technically need to be delayed by a render cycle, not a full ~16ms
Could see set some state here instead, then in response to that state change call the scrollIntoViewport?

Reply:
This is also pretty specific to calendars in a popover so it's is a bit weird to put in here. I guess this would technically be the same for _any_ component inside a popover that called scroll into view. Is there a way to disable scroll into view until popovers are positioned more generally?

Reply:
Well, the information for if a popover has finished positioning is [here](https://github.com/adobe/react-spectrum/blob/38a57d3360268fb0cb55c6b42b9a5f6f13bb57d6/packages/%40react-aria/overlays/src/usePopover.ts#L98), so I'd need to have scrollIntoViewport somehow receive that information + know that the target element is within a popover.

Reply:
Updated, had to create another piece of global state so the popover could inform the calendar cell when it was finished positioning and that it should let scrollIntoViewport happen now. I don't feel that great about that global state but didn't feel that just delaying it by a render cycle was really appropriate (feels more brittle and reliant on implementation), lemme know if you have any suggestions on getting rid of the global state